src/libs/Navigation/AppNavigator/AppNavigator.js

```diff
@@ -0,0 +1,190 @@
+
+    // Modal routes are presented above the content or in a specified way on larger screen widths
+    modalRoutes: modalRoutesPropTypes.isRequired,
+
+    // Whether the drawer should be open on init
+    isDrawerOpenByDefault: PropTypes.bool.isRequired,
+
+    ...windowDimensionsPropTypes,
+};
+
+const defaultProps = {
+    responsive: false,
+};
+
+const AppNavigator = (props) => {
+    const RootNavigator = RootStack;
+    return (
+        <RootNavigator.Navigator
+            mode="modal"
+        >
+            {props.authenticated
+                ? (
+
+                    // These are the protected screens and only accessible when an authToken is present.
+                    <>
+                        <RootNavigator.Screen
+                            name="Home"
+                            options={{
+                                headerShown: false,
+                                title: 'Expensify.cash',
+                            }}
+                        >
+                            {() => (
+                                <AppWrapper>
+                                    <Drawer.Navigator
+                                        openByDefault={props.isDrawerOpenByDefault}
+                                        drawerType={getNavigationDrawerType(props.isSmallScreenWidth)}
+                                        drawerStyle={getNavigationDrawerStyle(
+                                            props.windowWidth,
+                                            props.isSmallScreenWidth,
+                                        )}
+                                        sceneContainerStyle={styles.navigationSceneContainer}
+                                        drawerContent={() => {
+                                            const SidebarComponent = props.sidebarRoute.Component;
+                                            return (
+                                                <SidebarComponent />
+                                            );
+                                        }}
+                                    >
+                                        <Drawer.Screen
+                                            name="Loading"
+                                            options={{
+                                                cardStyle: styles.navigationScreenCardStyle,
+                                                headerShown: false,
+                                            }}
+                                        >
+                                            {() => {
+                                                const ReportScreen = props.mainRoutes[0].Component;
+                                                return props.isSmallScreenWidth ? <View /> : <ReportScreen />;
+                                            }}
+                                        </Drawer.Screen>
+                                        {_.map(props.mainRoutes, route => (
```

Comment:
imo all these different grouping of screens (mainRoutes, modalRoutes, etc.) should all be broken out into their own files. also dont need to pass everything as props, can simply define all the routes/props there.

Reply:
I'm going to skip this. While I like the idea of breaking this down further it's probably fine to just having Auth and Public screens to start.